### PR TITLE
Added all Docker machine operations for Photon Controller driver

### PR DESCRIPTION
**Pull request by faijaz**
Added the following Docker machine operations for Photon Controller driver: 
1. Start
2. Stop
3. Restart
4. Kill
5. Remove
6. Get Status

### CODE REVIEW

File: drivers/photoncontroller/photoncontroller.go [hunk 1]
```diff
@@ -271,29 +271,113 @@ func (d *Driver) PreCreateCheck() error {
...
+	fmt.Println("VM was restarted.")
+
+	return nil
 }
 
 func (d *Driver) Kill() error {
-	return NotLoadable{d.Name}
-}
+	client := d.getClient()
+	opTask, err := client.VMs.Stop(d.VMId)
```
Comment (AlainRoy) [inline drivers/photoncontroller/photoncontroller.go]: I don't understand what the difference between kill and stop are. This does a stop, but (unlike the stop command) doesn't wait. 

Can you explain what kill() is supposed to do? Maybe add a comment?

File: drivers/photoncontroller/photoncontroller.go [hunk 2]
```diff
@@ -271,29 +271,113 @@ func (d *Driver) PreCreateCheck() error {
...
+		return state.Paused, nil
+	}
+
+	return state.Error, nil
 }
 
 func (d *Driver) Remove() error {
-	return NotLoadable{d.Name}
+	client := d.getClient()
+	opTask, err := client.VMs.Delete(d.VMId)
```
Comment (AlainRoy) [inline drivers/photoncontroller/photoncontroller.go]: Delete() only works if you stopped the VM previously. Should this get the state and stop the VM if necessary?

File: drivers/photoncontroller/photoncontroller.go (hunk 1 above)
Comment (faijaz) [inline drivers/photoncontroller/photoncontroller.go]: It's essentially the same as stop.

File: drivers/photoncontroller/photoncontroller.go (hunk 2 above)
Comment (faijaz) [inline drivers/photoncontroller/photoncontroller.go]: Fixed it.

**Comment by AlainRoy** [comment]
Looks good to me!

**Pull request merged**